tests/phpunit/api/v4/Query/Api4SelectQueryTest.php

```diff
@@ -107,4 +107,21 @@ public function testInvalidSort(): void {
     }
   }
 
+  public function testInvalidPseudoConstantWithIN(): void {
+    $this->createTestRecord('Contact', [
+      'first_name' => uniqid(),
+      'last_name' => uniqid(),
+      'prefix_id:name' => 'Ms.',
+    ]);
+    $api = Request::create('Contact', 'get', [
+      'version' => 4,
+      'checkPermissions' => FALSE,
+      'select' => ['id'],
+      'where' => [['prefix_id:name', 'IN', ['Msssss.', 'Ms.']]],
+    ]);
+    $query = new Api4SelectQuery($api);
+    $result = $query->run();
+    $this->assertCount(0, $result);
```

Comment:
@MegaphoneJon megapoints for writing a unit test.
This test appears to:
1. Create a contact with `prefix_id:name = 'Ms.'`.
2. Search for contacts with `'prefix_id:name'` that includes `'Ms.'` and some other irrelevant stuff.
3. Expect to not find any contacts, even though it just created one that clearly matches the search criteria.
That seems wrong to me. What am I missing?

Also, this `Api4SelectQueryTest` class is testing the guts of the `DaoEntity::get()` action. That's IMO too low-level for this test which really ought to be testing `Contact::get()` directly. Maybe move this to `\api\v4\Action\ContactGetTest`?

Reply:
Ah, turns out the test fails as written but would pass if it expected 1 result. Which is what I would expect too :)